From 8f1a67458dfd5c4356a835da1fd9d130f5755de9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michel=20D=C3=A4nzer?= <mdaenzer@redhat.com>
Date: Fri, 23 Jun 2023 18:30:07 +0200
Subject: [PATCH 066/105] composite: Expose
 CompositeIsImplicitRedirectException

Make it usable by code outside of the composite layer.
---
 composite/compositeext.h | 2 ++
 composite/compwindow.c   | 8 ++++----
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/composite/compositeext.h b/composite/compositeext.h
index 5aad0735e..27ce7ce40 100644
--- a/composite/compositeext.h
+++ b/composite/compositeext.h
@@ -41,6 +41,8 @@ extern _X_EXPORT Bool CompositeRegisterImplicitRedirectionException(ScreenPtr pS
 
 
 extern _X_EXPORT Bool compIsAlternateVisual(ScreenPtr pScreen, XID visual);
+Bool CompositeIsImplicitRedirectException(ScreenPtr pScreen,
+                                          XID parentVisual, XID winVisual);
 extern _X_EXPORT RESTYPE CompositeClientWindowType;
 
 #endif                          /* _COMPOSITEEXT_H_ */
diff --git a/composite/compwindow.c b/composite/compwindow.c
index 9a651636e..0bdcd8aa7 100644
--- a/composite/compwindow.c
+++ b/composite/compwindow.c
@@ -339,9 +339,9 @@ compIsAlternateVisual(ScreenPtr pScreen, XID visual)
     return FALSE;
 }
 
-static Bool
-compIsImplicitRedirectException(ScreenPtr pScreen,
-                                XID parentVisual, XID winVisual)
+Bool
+CompositeIsImplicitRedirectException(ScreenPtr pScreen,
+                                     XID parentVisual, XID winVisual)
 {
     CompScreenPtr cs = GetCompScreen(pScreen);
     int i;
@@ -362,7 +362,7 @@ compImplicitRedirect(WindowPtr pWin, WindowPtr pParent)
         XID winVisual = wVisual(pWin);
         XID parentVisual = wVisual(pParent);
 
-        if (compIsImplicitRedirectException(pScreen, parentVisual, winVisual))
+        if (CompositeIsImplicitRedirectException(pScreen, parentVisual, winVisual))
             return FALSE;
 
         if (winVisual != parentVisual &&
-- 
2.45.2

